roles/openshift_repos/tasks/main.yaml

```diff
@@ -7,44 +7,51 @@
-  when: not openshift.common.is_containerized | bool
-
-- name: Create any additional repos that are defined
-  template:
-    src: yum_repo.j2
-    dest: /etc/yum.repos.d/openshift_additional.repo
-  when: openshift_additional_repos | length > 0 and not openshift.common.is_containerized | bool
-  notify: refresh cache
-
-- name: Remove the additional repos if no longer defined
-  file:
-    dest: /etc/yum.repos.d/openshift_additional.repo
-    state: absent
-  when: openshift_additional_repos | length == 0 and not openshift.common.is_containerized | bool
-  notify: refresh cache
-
-- name: Configure origin gpg keys if needed
-  copy:
-    src: origin/gpg_keys/openshift-ansible-CentOS-SIG-PaaS
-    dest: /etc/pki/rpm-gpg/
-  notify: refresh cache
-  when: ansible_os_family == "RedHat" and ansible_distribution != "Fedora"
-        and openshift_deployment_type == 'origin'
-        and not openshift.common.is_containerized | bool
-        and openshift_enable_origin_repo | default(true) | bool
-
-- name: Configure origin yum repositories RHEL/CentOS
-  copy:
-    src: origin/repos/openshift-ansible-centos-paas-sig.repo
-    dest: /etc/yum.repos.d/
-  notify: refresh cache
-  when: ansible_os_family == "RedHat" and ansible_distribution != "Fedora"
-        and openshift_deployment_type == 'origin'
-        and not openshift.common.is_containerized | bool
-        and openshift_enable_origin_repo | default(true) | bool
+- name: openshift_repos detect ostree
+  stat:
+    path: /run/ostree-booted
+  register: ostree_booted
+
+- block:
+  - assert:
+      that: openshift_deployment_type in known_openshift_deployment_types
+
+  - name: Ensure libselinux-python is installed
+    package: name=libselinux-python state=present
+
+  - name: Create any additional repos that are defined
+    template:
+      src: yum_repo.j2
+      dest: /etc/yum.repos.d/openshift_additional.repo
+    when:
+    - openshift_additional_repos | length > 0
+    notify: refresh cache
+
+  - name: Remove the additional repos if no longer defined
+    file:
+      dest: /etc/yum.repos.d/openshift_additional.repo
+      state: absent
+    when:
+    - openshift_additional_repos | length == 0
+    notify: refresh cache
+
+  - name: Configure origin gpg keys if needed
+    copy:
+      src: origin/gpg_keys/openshift-ansible-CentOS-SIG-PaaS
+      dest: /etc/pki/rpm-gpg/
+    notify: refresh cache
+    when:
+    - ansible_os_family == "RedHat" and ansible_distribution != "Fedora"
+    - openshift_deployment_type == 'origin'
+    - openshift_enable_origin_repo | default(true) | bool
+
+  - name: Configure origin yum repositories RHEL/CentOS
+    copy:
+      src: origin/repos/openshift-ansible-centos-paas-sig.repo
+      dest: /etc/yum.repos.d/
+    notify: refresh cache
+    when:
+    - ansible_os_family == "RedHat" and ansible_distribution != "Fedora"
+    - openshift_deployment_type == 'origin'
+    - openshift_enable_origin_repo | default(true) | bool
```

Comment:
I'd condense these two copy tasks into a single task:
```
- name: Configure origin gpg keys if needed
  copy:
    src: origin/gpg_keys/openshift-ansible-CentOS-SIG-PaaS
    dest: /etc/pki/rpm-gpg/
  notify: refresh cache
  with_items:
  - src: origin/gpg_keys/openshift-ansible-CentOS-SIG-PaaS
    dest: /etc/pki/rpm-gpg/
  - src: origin/repos/openshift-ansible-centos-paas-sig.repo
    dest: /etc/yum.repos.d/
  when:
  - ansible_os_family == "RedHat"
  - ansible_distribution != "Fedora"
  - openshift_deployment_type == 'origin'
  - openshift_enable_origin_repo | default(true) | bool
```

I'm not entirely sure that we need to keep `ansible_os_family == "RedHat"`, but it doesn't hurt to leave it in place.